services/keybase/keybase-pgp.service.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+const KeybaseProfile = require('./keybase-profile')
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+
+const keyFingerprintSchema = Joi.object({
+  status: Joi.object({
+    code: nonNegativeInteger.required(),
+  }).required(),
+  them: Joi.array()
+    .items(
+      Joi.object({
+        public_keys: {
+          primary: {
+            key_fingerprint: Joi.string()
+              .hex()
+              .required(),
+          },
+        },
+      })
+        .required()
+        .allow(null)
+    )
+    .min(0)
+    .max(1),
+}).required()
+
+module.exports = class KeybasePGP extends KeybaseProfile {
+  static get apiVersion() {
+    return '1.0'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'keybase/pgp',
+      pattern: ':username',
+    }
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'pgp',
+      color: 'informational',
+    }
+  }
+
+  async handle({ username }) {
+    const options = {
+      form: {
+        usernames: username,
+        fields: 'public_keys',
+      },
+    }
+
+    const data = await this.fetch({
+      schema: keyFingerprintSchema,
+      options,
+    })
+
+    if (data.status.code !== 0) {
+      return {
+        message: 'invalid username',
+        color: 'critical',
+      }
+    }
+
+    if (data.them.length === 0 || !data.them[0]) {
+      return {
+        message: 'profile not found',
+        color: 'critical',
+      }
+    }
+
+    const primaryKey = data.them[0].public_keys.primary
```

Comment:
with the above change to leverage the base `transform` function, this line can be changed to:
```js
const { user } = this.transform({ data })
const primaryKey = user.public_keys.primary
```